microsetta_public_api/config.py

```diff
@@ -1,225 +1,388 @@
+        }
+
+
+class LegacySchema(SchemaBase):
+    def __init__(self):
+        self.alpha_kw = 'alpha_resources'
+        self.taxonomy_kw = 'table_resources'
+        self.pcoa_kw = 'pcoa'
+        self.metadata_kw = 'metadata'
+
+    def schema(self):
+        return {
+            "type": "object",
+            "properties": {
+                self.alpha_kw: alpha_group_schema,
+                self.taxonomy_kw: taxonomy_group_schema,
+                self.pcoa_kw: pcoa_group_schema,
+                self.metadata_kw: metadata_schema,
+            }
+        }
+
+
+class CompatibilitySchema(SchemaBase):
+    def __init__(self):
+        self.old_alpha_kw = 'alpha_resources'
+        self.old_taxonomy_kw = 'table_resources'
+        self.old_pcoa_kw = 'pcoa'
+        self.old_metadata_kw = 'metadata'
+        self.new_alpha_kw = '__alpha__'
+        self.new_taxonomy_kw = '__taxonomy__'
+        self.new_pcoa_kw = '__pcoa__'
+        self.new_metadata_kw = '__metadata__'
+
+    def element_map(self):
+        return {
+            self.old_alpha_kw: AlphaElement,
+            self.old_taxonomy_kw: TaxonomyElement,
+            self.old_pcoa_kw: PCOAElement,
+            self.old_metadata_kw: MetadataElement,
+            self.new_alpha_kw: AlphaElement,
+            self.new_taxonomy_kw: TaxonomyElement,
+            self.new_pcoa_kw: PCOAElement,
+            self.new_metadata_kw: MetadataElement,
+        }
+
+    def schema(self):
+        return {
+            "type": "object",
+            "properties": {
+                self.old_alpha_kw: alpha_group_schema,
+                self.old_taxonomy_kw: taxonomy_group_schema,
+                self.old_pcoa_kw: pcoa_group_schema,
+                self.old_metadata_kw: metadata_schema,
+                "datasets":
+                    {
+                        "type": "object",
+                        "properties": {
+                            self.new_metadata_kw: metadata_schema,
+                        },
+                        "additionalProperties":
+                            {
+                                "type": "object",
+                                "properties": {
+                                    self.new_alpha_kw: alpha_group_schema,
+                                    self.new_taxonomy_kw:
+                                        taxonomy_group_schema,
+                                    self.new_pcoa_kw: pcoa_group_schema,
+                                },
+                                "additionalProperties": False,
+                            }
+                    },
+            }
+        }
+
+
+def create_literal_element(literal):
+    class LiteralElement(Element, literal):
+        def accept(self, visitor):
+            pass
+    return LiteralElement
+
+
+class ElementFactory:
+
+    @staticmethod
+    def get_element(obj):
+        if obj is None:
+            return None
+        if isinstance(obj, bool):
+            return obj
```

Comment:
Why don't you need to create_literal_element on bools?

Reply:
You cannot inherit `bool` in python. So we could not make an `Element` that acts just like `bool` but has its own `accept` method. The `bool` is probably part of an object or list that actually knows what to do with it anyways, so might as well just keep it as a bool.